### PR TITLE
Costmap_2d Static & Inflation plugin layer port & Removing Boost dependencies 

### PR DESCRIPTION
**Pull request by bpwilcox**
In this PR, the static layer & inflation layer plugins have been ported and compile properly. After a manual inspection of the plugins, they appear to work fine and generate reasonable results, though some additional work needs to be done in order to incorporate the plugins properly in the Costmap2DROS class (particularly with parameters). 

Some intial work on the observation buffer was done but was blocked from completion until tf2_sensor_msgs is also ported (will file an issue). 

Also, Boost dependencies have been removed from costmap_2d.

### CODE REVIEW

File: src/libs/costmap_2d/package.xml [hunk 1]
```diff
@@ -42,6 +42,7 @@
   <depend>xmlrpcpp</depend>
   <depend>pluginlib</depend>
   <depend>libpcl-all-dev</depend>
+ 	<depend>message_filters</depend>
```
Comment (crdelsey) [inline src/libs/costmap_2d/package.xml]: Spacing is off

File: src/libs/costmap_2d/plugins/static_layer.cpp [hunk 2]
```diff
@@ -55,72 +55,91 @@ StaticLayer::StaticLayer() : dsrv_(NULL) {}
...
-  }
+  } */
 }
 
 void StaticLayer::onInitialize()
 {
-  ros::NodeHandle nh("~/" + name_), g_nh;
+  auto nh = rclcpp::Node::make_shared(name_);
+  rclcpp::Node::SharedPtr g_nh;
+  auto parameters_client = std::make_shared<rclcpp::SyncParametersClient>(nh);
```
Comment (crdelsey) [inline src/libs/costmap_2d/plugins/static_layer.cpp]: You can just use the parameter_client built into the node.

File: src/libs/costmap_2d/include/costmap_2d/testing_helper.h [hunk 3]
```diff
@@ -59,9 +61,13 @@ unsigned int countValues(costmap_2d::Costmap2D & costmap, unsigned char value, b
 
 void addStaticLayer(costmap_2d::LayeredCostmap & layers, tf2_ros::Buffer & tf)
 {
+
```
Comment (crdelsey) [inline src/libs/costmap_2d/include/costmap_2d/testing_helper.h]: No need for all these blank lines.

File: src/libs/costmap_2d/plugins/static_layer.cpp [hunk 4]
```diff
@@ -62,72 +63,68 @@ StaticLayer::~StaticLayer()
 
 void StaticLayer::onInitialize()
 {
+
```
Comment (crdelsey) [inline src/libs/costmap_2d/plugins/static_layer.cpp]: Delete blank lines

File: src/libs/costmap_2d/CMakeLists.txt [hunk 5]
```diff
@@ -80,21 +78,18 @@ set(dependencies
...
-target_link_libraries(nav2_costmap_2d
-  ${Boost_LIBRARIES}
-)
-
 add_library(layers
   plugins/inflation_layer.cpp
 
-  # TODO(bpwilcox): port additional plugin layers
-  #plugins/static_layer.cpp
+	#	TODO(bpwilcox): port additional plugin layers
```
Comment (crdelsey) [inline src/libs/costmap_2d/CMakeLists.txt]: Spacing is off

File: src/libs/costmap_2d/plugins/static_layer.cpp (hunk 2 above)
Comment (bpwilcox) [reply src/libs/costmap_2d/plugins/static_layer.cpp]: I made an issue for this in #119. I was planning to make this replacement all in one PR.

File: src/libs/costmap_2d/package.xml (hunk 1 above)
Comment (bpwilcox) [reply src/libs/costmap_2d/package.xml]: I'm not sure why my formatting keeps getting off on these files (package.xml & CMakeLists.txt), I have to keep fixing this. Does opening it in gedit somehow autofix the formatting?

Comment (crdelsey) [reply src/libs/costmap_2d/package.xml]: I wouldn't think so, but I'd recommend a 'real' editor as you main choice anyhow. Then you can configure it to do the right thing.

**Pull request merged**